### PR TITLE
Support move movement.

### PR DESCRIPTION
**Pull request by ArthurSonzogni**
- Record whether the mouse moved. Suppress the previous alternative implementation.
- Record mouse.control modifier.

Fixed:https://github.com/ArthurSonzogni/FTXUI/issues/791

### CODE REVIEW

**Comment by ArthurSonzogni** [comment]
Hello @psychon,
Could you please take a look? Does this improve the situation?

**Comment by psychon** [comment]
> Does this improve the situation?

TL;DR: Yup, it does. I can't spot anything really wrong that can be fixed in ftxui.

Let's look at `ftxui_example_print_key_press` again:

- Open a new urxvt instance, run this, move the mouse around reports things like `mouse_none_pressed_control_shift_meta(95,24)`; this is a bug in urxvt that I found yesterday and I am still trying to report upstream (urxvt's code reports "button -1 was last pressed" and that messes up everything)
- After any mouse press, things become `mouse_left_moved(61, 6)` or `mouse_right_moved(63, 5)` or something like that (same for middle button and wheels)
- Click-and-drag produces stuff like
  -  `mouse_left_pressed(...)`
  -  `mouse_left_moved(...)`
  -  `mouse_left_moved(...)`
  -  `mouse_left_released(...)`
- Ctrl + mouse click moves fine, e.g. `mouse_left_pressed_control`
- Alt + mouse click moves fine, e.g. `mouse_left_pressed_meta`
- Ctrl + Alt + mouse clicks works, too
- Shift + mouse click selects words and is not detected

Basically the same behaviour with xterm, except:
- Things work fine even before the first mouse click
- Moves already report pressed modifiers (with urxvt I always get the modifiers active at the time of the last mouse click)
- Ctrl + click is caught by xterm and not reported

The `dropdown` example works fine.

I'm out of other ideas to try.

**Comment by ArthurSonzogni** [comment]
Thank you very much!

**Comment by psychon** [comment]
Thank *you* for fixing this so quickly
